pkg/pbq/store/wal/stores.go

```diff
@@ -35,13 +37,19 @@ type walStores struct {
 	maxBatchSize int64
 	// syncDuration timeout to sync to store
 	syncDuration time.Duration
+	pipelineName string
+	vertexName   string
+	replicaIndex int32
 }
 
-func NewWALStores(opts ...Option) store.StoreProvider {
+func NewWALStores(vertexInstance *dfv1.VertexInstance, opts ...Option) store.StoreProvider {
```

Comment:
If it's only for metrics purpose, is it good to only pass a mapped labels?

Reply:
No strong opinion though.

Reply:
That's a good point. I tend to think it makes sense that when we create a WAL store, we specify under which vertex instance we are creating it. It's mostly used for emitting metrics atm but I guess could be other usages in the future.

Reply:
Let's keep it as-is for consistency? I think in other places too we are passing `*dfv1.VertexInstance` instead of a map.